pyro/contrib/gp/models/svgp.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import absolute_import, division, print_function
+
+import torch
+from torch.autograd import Variable
+from torch.distributions import constraints, transform_to
+from torch.nn import Parameter
+
+import pyro
+import pyro.distributions as dist
+from pyro.distributions.util import matrix_triangular_solve_compat
+
+from .model import Model
+
+
+class SparseVariationalGP(Model):
+    """
+    Sparse Variational Gaussian Process module.
+
+    References
+
+    [1] `Scalable variational Gaussian process classification`,
+    James Hensman, Alexander G. de G. Matthews, Zoubin Ghahramani
+
+    :param torch.autograd.Variable X: A tensor of inputs.
+    :param torch.autograd.Variable y: A tensor of outputs for training.
+    :param pyro.contrib.gp.kernels.Kernel kernel: A Pyro kernel object.
+    :param torch.Tensor Xu: An inducing-point parameter.
+    :param pyro.contrib.gp.likelihoods.Likelihood likelihood: A likelihood module.
+    :param float jitter: An additional jitter to help stablize Cholesky decomposition.
+    """
+    def __init__(self, X, y, kernel, Xu, likelihood, jitter=1e-6):
+        super(SparseVariationalGP, self).__init__()
+        self.X = X
+        self.y = y
+        self.kernel = kernel
+        self.num_data = self.X.size(0)
+
+        self.Xu = Parameter(Xu)
+        self.num_inducing = self.Xu.size(0)
+
+        self.likelihood = likelihood
+
+        self.jitter = Variable(self.X.data.new([jitter]))
+
+    def model(self):
+        self.set_mode("model")
+
+        kernel = self.kernel
+        likelihood = self.likelihood
+        Xu = self.get_param("Xu")
+
+        Kuu = kernel(Xu) + self.jitter.expand(self.num_inducing).diag()
+        Kuf = kernel(Xu, self.X)
+        Luu = Kuu.potrf(upper=False)
+
+        zero_loc = Variable(Xu.data.new([0])).expand(self.num_inducing)
+        # TODO: use scale_tril=Luu
+        u = pyro.sample("u", dist.MultivariateNormal(loc=zero_loc, covariance_matrix=Kuu))
+
+        # p(f|u) ~ N(f|mf, Kf)
+        # mf = Kfu @ inv(Kuu) @ u; Kf = Kff - Kfu @ inv(Kuu) @ Kuf
+        pack = torch.cat((u.unsqueeze(1), Kuf), dim=1)
```

Comment:
Right now, `torch.trtrs` does not support grad or cuda, so we have to use `.inverse()` to solve. Combining will be much faster (because we don't have to take inverse two times). In case we use `trtrs`, the performance is similar.